### ISSUE TITLE
`+client.js`

### ISSUE THREAD

**Issue by brillout** [vikejs/vike#1468]
Implement https://vike.dev/client. (It currently only works for HTML-only pages.)

Let us know if this missing feature is a blocker for you.

**Comment by ChristophP** [comment]
We actually have a need for this.

At Burda Forward we do not do full but partial hydration.
So it makes sense for us to

```
// /renderer/+config.ts
export default {
  meta: {
    Page: { server: true, client: false },
  }
}
```

In this case it seems that the `+onRenderClient` hook is not triggered due to the missing client-side `Page`.

Currently in the v0 design we use hydration logic similar to this
```ts
// dummy.page.client.ts
import { hydrateStatic } from '@renderer/core/client';
import { componentsToHydrate } from '@framework/templates/default-template/some-brand';
import '@framework/templates/default-template/some-brand/style.css';

async function onRenderClient(pageContext): Promise<void> {
  console.log('B');
  /* Read server data on client-side, after 'passToClient'
   * https://vike.dev/passToClient */
  const { islands } = pageContext; // islands is a bag of all the components that need to be hydrated along with the divs and their ids
  hydrateIslands(islands, componentsToHydrate);
}
export default onRenderClient;
```

Doing the same in `client.ts` would be nice as long as it is possible to access the pageContext there (which we need to retrieve the islands)

**Comment by brillout** [comment]
@ChristophP `+client` is global and thus doesn't have access to `pageContext` (since `pageContext` is tied to the process of a page being rendered).

**Comment by ChristophP** [comment]
@brillout I see, so it's more suitable to set up tracking, error logging, consent layer etc

**Comment by brillout** [comment]
Exactly. I've improved the docs to reflect that (https://github.com/vikejs/vike/commit/c669d8734fc4aeb34ad00401f0d5a020d5a89b3f).

**Comment by Dubzer** [comment]
I think this feature can be useful for handling theming, to avoid flickering while hydrating

**Comment by brillout** [comment]
Pre-released as `0.4.171-commit-6e37128`.
